Tests/Shared.Specs/FunctionExceptionAssertionSpecs.cs

```diff
@@ -0,0 +1,545 @@
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act / Assert
+            //-----------------------------------------------------------------------------------------------------------
+            f.Should().Throw<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void When_subject_does_not_throw_expected_exception_it_should_fail()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            Func<int> f = () => throw new ArgumentNullException();
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => f.Should().Throw<InvalidCastException>();
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action.Should().Throw<XunitException>()
+                .WithMessage("Expected*InvalidCastException*but*ArgumentNullException*");
+        }
+
+        [Fact]
+        public void When_subject_does_not_throw_any_exception_it_should_fail()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            Func<int> f = () => 12;
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => f.Should().Throw<InvalidCastException>("that's what I {0}", "said");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action.Should().Throw<XunitException>()
+                .WithMessage("Expected*InvalidCastException*that's what I said*but*no exception*");
+        }
+
+        [Fact]
+        public void When_subject_does_not_throw_exception_and_that_was_expected_it_should_succeed_then_continue_assertion()
```

Comment:
The method name does not reflect the assertion

Reply:
👍